aten/src/ATen/native/SpectralOps.cpp

```diff
@@ -17,23 +17,212 @@
 #include <cmath>
 
 namespace at { namespace native {
+static inline Tensor _fft(
+    const Tensor &self, int64_t signal_ndim, bool complex_input,
+    const bool complex_output, bool inverse, IntArrayRef signal_sizes,
+    fft_norm_mode normalization, bool onesided);
+
+namespace {
+
+ScalarType promote_type(ScalarType type, bool require_complex) {
+  if (at::isComplexType(type)) {
+    return type;
+  }
+  // Promote integral to double
+  if (!at::isFloatingType(type)) {
```

Comment:
For consistency with PyTorch should this instead promote to the default complex type?

Reply:
Done.